build/templates/E2E_Tests_On_Windows.yml

```diff
@@ -0,0 +1,108 @@
+  inputs:
+    scriptName: "$(System.DefaultWorkingDirectory)\\artifacts\\EndToEnd\\scripts\\SetupMachine.ps1"
+
+- task: PowerShell@1
+  displayName: "InstallNuGetVSIX.ps1"
+  inputs:
+    scriptName: "$(System.DefaultWorkingDirectory)\\artifacts\\EndToEnd\\scripts\\InstallNuGetVSIX.ps1"
+    arguments: "-NuGetDropPath $(System.DefaultWorkingDirectory)\\artifacts\\$(VsixPublishDir) -FuncTestRoot $(System.DefaultWorkingDirectory)\\artifacts -NuGetVSIXID $(NuGetVsixId) -ProcessExitTimeoutInSeconds 180 -VSVersion 16.0"
+    failOnStandardError: "false"
+
+- task: PowerShell@1
+  displayName: "Collect VS Logs"
+  inputs:
+    scriptType: "inlineScript"
+    inlineScript: |
+      Invoke-WebRequest -Uri $(CollectExeUrl) -OutFile $(System.DefaultWorkingDirectory)\\VSCollect.exe
+      $(System.DefaultWorkingDirectory)\\VSCollect.exe -zip:$(System.DefaultWorkingDirectory)\\artifacts\\EndToEnd\\e2e-collectlogs.zip
+  condition: "failed()"
+
+- task: PowerShell@1
+  displayName: "RunFunctionalTests.ps1 (stop on error)"
+  timeoutInMinutes: 75
+  continueOnError: "true"
+  inputs:
+    scriptName: "$(System.DefaultWorkingDirectory)\\artifacts\\EndToEnd\\scripts\\RunFunctionalTests.ps1"
+    arguments: "-PMCCommand $(EndToEndTestCommandToRunPart) -PMCLaunchWaitTimeInSecs 30 -EachTestTimoutInSecs 600 -NuGetDropPath $(System.DefaultWorkingDirectory)\\artifacts -FuncTestRoot $(System.DefaultWorkingDirectory)\\artifacts -RunCounter $(Build.BuildNumber) -VSVersion 16.0"
+  condition: "eq(variables['IsOfficialBuild'], 'true')"
+
+- task: PowerShell@1
+  displayName: "RunFunctionalTests.ps1 (continue on error)"
+  timeoutInMinutes: 75
+  continueOnError: "true"
+  inputs:
+    scriptName: "$(System.DefaultWorkingDirectory)\\artifacts\\EndToEnd\\scripts\\RunFunctionalTests.ps1"
+    arguments: "-PMCCommand $(EndToEndTestCommandToRunPart) -PMCLaunchWaitTimeInSecs 30 -EachTestTimoutInSecs 600 -NuGetDropPath $(System.DefaultWorkingDirectory)\\artifacts -FuncTestRoot $(System.DefaultWorkingDirectory)\\artifacts -RunCounter $(Build.BuildNumber) -VSVersion 16.0"
+  condition: "not(eq(variables['IsOfficialBuild'], 'true'))"
+
+- task: PublishTestResults@2
+  displayName: "Publish Test Results"
+  inputs:
+    testRunner: "JUnit"
+    testResultsFiles: "*.xml"
+    searchFolder: "$(System.DefaultWorkingDirectory)\\testresults"
+    mergeTestResults: "true"
+    testRunTitle: "NuGet.Client EndToEnd Tests On Windows"
+  condition: "succeededOrFailed()"
+
+- task: PowerShell@1
+  displayName: "Initialize Git Commit Status on GitHub"
+  inputs:
+    scriptType: "inlineScript"
+    arguments: "-VstsPersonalAccessToken $(VstsPersonalAccessToken)"
+    inlineScript: |
+      . $(Build.Repository.LocalPath)\\scripts\\utils\\PostGitCommitStatus.ps1
+      SetCommitStatusForTestResult -PersonalAccessToken $(NuGetLurkerPersonalAccessToken) -VstsPersonalAccessToken $(VstsPersonalAccessToken) -CommitSha $(Build.SourceVersion) -TestName "EndToEnd Tests On Windows"
```

Comment:
You need to modify the `PostGitCommitStatus` script so that:
1. during the Initialize job, it creates two End To End Test statuses, and each of the E2E jobs report their status to a different github status.

Reply:
Thanks for pointing out. I've fixed the script `PostGitCommitStatus`.